src/porepy/models/solution_strategy.py

```diff
@@ -511,7 +511,7 @@ def after_simulation(self) -> None:
     def check_convergence(
         self,
         nonlinear_increment: np.ndarray,
-        residual: np.ndarray,
+        residual: Optional[np.ndarray],
```

Comment:
I don't think optional arguments should appear before required arguments in a signature.

Reply:
`residual` is still required and not optional. The typing just indicates that the argument could have value `None`, which is here indeed the case if the residual is not computed for the computation of the norm.

Reply:
Sorry, you're quite right!